### PR TITLE
add golangci-lint as linter

### PR DESCRIPTION
**Pull request by PascalBourdier**
give us some advices

### CODE REVIEW

**Comment by PascalBourdier** [comment]
@argvk @yannh : is it ok for you ? or you don't like this linter ?

File: .circleci/config.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ jobs:
       docker_layer_caching: true
     steps:
       - checkout
+      - run: make lint
```
Comment (argvk) [inline .circleci/config.yml]: this should be a different job ideally, can you instead add a new `lint` job with the `make lint` step and add to the `workflow.flow.jobs` lower in the file

Comment (PascalBourdier) [reply .circleci/config.yml]: Ok, I fix it !

**Comment by argvk** [comment]
Thanks!

**Pull request merged**